<?xml version="1.0"?>

<!-- Used to test the combined outputs of all the controllers. -->

<!-- Written by Joshua Petrin -->

<launch>

 <!-- Core PNR -->
 <param name="/pnr_core/uswift_vector_scale" type="double" value="25.0" />
 <param name="/pnr_core/roomba_vector_scale" type="double" value="0.2" />
 <param name="/pnr_core/roomba_angular_scale" type="double" value="1.5" />
 <node pkg="pnr_core" type="pnr_core.py" name="pnr_core_node" output="screen" />

 <!-- Throttlers -->
 <node name="vector_write_throttler" type="throttle" pkg="topic_tools" args="messages /pnr_swiftpro/vector_write_t 8 /pnr_swiftpro/vector_write" />
 <node name="cmd_vel_throttler" type="throttle" pkg="topic_tools" args="messages /cmd_vel_t 8 /cmd_vel" />

 <!-- The teleop keyboard node; included here for debugging -->
 <node pkg="teleop_twist_keyboard" name="teleop_controller" type="teleop_twist_keyboard.py" output="screen">
  <remap from="cmd_vel" to="teleop_keyboard/twist" />
  <param name="speed" value="0.5" />
 </node>

 <!-- All of the joysticks! If one node is not connected to its
      counterpart, it will keep listening until it finds it (i.e. when
      it is plugged in to the pi) -->
 <node pkg="joy" type="joy_node" name="xbox_joy" output="screen">
  <remap from="joy" to="xbox/joy" />
  <param name="dev" type="string" value="/dev/xbox" />
  <param name="deadzone" value="0.12" />
 </node>
 <node pkg="joy" type="joy_node" name="xbox_adaptive_joy" output="screen">
  <remap from="joy" to="xbox_adaptive/joy" />
  <param name="dev" type="string" value="/dev/xbox_adaptive" />
  <param name="deadzone" value="0.05" />
 </node>
 <node pkg="joy" type="joy_node" name="quadstick_joy" output="screen">
  <remap from="joy" to="quadstick/joy" />
  <param name="dev" type="string" value="/dev/quadstick" />
  <param name="deadzone" value="0.05" />
 </node>
 <node pkg="spacenav_node" type="spacenav_node" name="$(anon spacenav_node)" output="screen">
  <param name="zero_when_static" value="true"/>
  <param name="static_count_threshold" value="30"/>
  <param name="deadzone" value="0.05" />
 </node>
</launch>
